addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -132,6 +132,8 @@ def portal_my_opportunities(self, page=1, date_begin=None, date_end=None, sortby
 
         if date_begin and date_end:
             domain += [('create_date', '>', date_begin), ('create_date', '<=', date_end)]
+        if filterby in ('today', 'week', 'overdue') and any(group.name == 'Portal' for group in request.env.user.groups_id):
```

Comment:
yes, as a general rule, it's rare that we match records on name (and especially rarer if the name has no unicity constraint and if the name is translatable (eg. here if someone didn't english it would fail))

reth-odoo checks if we have the group using XML ID, usually we would either do that or directly check if we have access right over the model or records